### PR TITLE
Document zarith 1.12 version bump.

### LINKED ISSUE #15496: Minimal zarith version for Coq 8.16.
Since the integration of #15483, the lower bound version of the Coq dependency `zarith` is untested. We should (quickly) decide whether we want to officially drop support for zarith 1.10 (which triggers a bug in bedrock2), and update the Docker file so that the minimally supported version is tested. If we do drop support for zarith 1.10, we should update the documentation and write a changelog entry. If we do not drop support, then we should just inform the users that they might hit some bugs with this version and that upgrading can fix them.

Note that it is possible to test zarith 1.10 in the switch that doesn't build bedrock2 and a more recent zarith version in the switch that does.

FWIW the Nix CI currently tests zarith 1.12 as well.

### PR DESCRIPTION
**Pull request by Zimmi48**
Fixes / closes #15496

### CODE REVIEW

**Comment by ppedrot** [comment]
AFAICT there are still remaining mentions of zarith 1.10.
```
.github/workflows/ci-macos.yml:          opam install -j "$NJOBS" ocamlfind${FINDLIB_VER} ounit lablgtk3-sourceview3 zarith.1.10 dune.2.8.5
coq-core.opam:  "zarith" {>= "1.10"}
coq.opam.docker:  "zarith"    { >= "1.10" }
dune-project:  (zarith (>= 1.10))
tools/configure/configure.ml:    if zarith_version_int >= [1;10;0] then
tools/configure/configure.ml:      die ("Zarith version 1.10 is required, you have " ^ zarith_version)
```

**Comment by Zimmi48** [comment]
Right, this is more than doc. I'll take care of these.

**Comment by Zimmi48** [comment]
That being said, this shows that zarith 1.10 was actually still tested in CI (but only on macOS).

File: doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- **Changed:**
+  Minimal supported zarith version is now 1.12
```
Comment (jfehrle) [inline doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst:2]:
```suggestion
  Minimum supported zarith version is now 1.12
```

Not the right word.  Could say "Lowest", too.

Comment (Alizter) [reply doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst:2]: Could you elaborate why you don't think it is right? It sounds fine to my ear.

Comment (ppedrot) [reply doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst:2]: (Both words also have a mathematical use which is correct in this context.)

Comment (Zimmi48) [reply doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst]: Despite the debate, I have committed the suggestion.

Comment (jfehrle) [reply doc/changelog/11-infrastructure-and-dependencies/15483-bump-zarith-doc.rst]: "Minimum", meaning "the least or smallest amount or quantity possible, attainable, or **required**" (Oxford) is precisely what's needed here.  "Minimal", meaning "of a minimum amount, quantity, or degree; **negligible**" (Oxford) is a bit off.  Certainly "negligible" is not what's intended.  I think mathematical use is not so relevant here, this section is not at all mathematical.  IMHO.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by JasonGross** [comment]
Is there any chance the version dependency could be dropped to 1.11?  This broke my Coq packages.  I can backport zarith 1.12 if it's too much work.

**Comment by Zimmi48** [comment]
How come your packages broke? I checked that zarith 1.12 was available in the latest Ubuntu LTS and Debian before proceeding to do this bump. Anyway, there is no reason why zarith 1.11 wouldn't be suitable, so feel free to open a PR amending this version bump (but make sure to also change at least one of the switches in the Docker image so that it properly tests the minimum required version).

**Comment by JasonGross** [comment]
They broke because I backported 1.11 before 1.12 was packaged anywhere, so my packages for older-than-newest-LTS are still using 1.11.

Do I need to rebuild the docker image somehow?

**Pull request merged**